### DISCUSSION TITLE
How to use LLVMValueRef.CreateConstIntOfString

### DISCUSSION THREAD

**Discussion by TrueTheos** [Microsoft/LLVMSharp#219] [Q&A] [answered]
```
public (Context, LLVMValueRef) VisitNumber(Context ctx, NumberAST expr)
{
    LLVMTypeRef typeRef = LLVMTypeRef.Double;

    return new(ctx, LLVMValueRef.CreateConstIntOfString(typeRef, expr.Value, 10));
}
```
(NumberAST has a variable `string Value` which contains `"10"`)

`var res = VisitNumber(ctx, new("10")).Item2; `
and the `res` is:`{i0 0}`

Is this correct result or is it wrong?

**Comment by tannergooding** [comment, marked as answer]
Answered this on Discord.

The consideration is that you're using `CreateConstIntOfString` with `LLVMTypeRef.Double`, when that is not an integer type.

You need to use `CreateConstRealOfString` instead.
